src/DBConnection.cpp

```diff
@@ -240,17 +240,16 @@ bool DBConnection::Close()
       }
 
       // Provides a progress dialog with indeterminate mode
-      wxGenericProgressDialog pd(title.Translation(),
-                                 XO("This may take several seconds").Translation(),
-                                 300000,     // range
-                                 nullptr,    // parent
-                                 wxPD_APP_MODAL | wxPD_ELAPSED_TIME | wxPD_SMOOTH);
+      using namespace BasicUI;
+      auto pd = MakeGenericProgress({},
+         title, XO("This may take several seconds"));
+      wxASSERT(pd);
 
       // Wait for the checkpoints to end
       while (mCheckpointPending || mCheckpointActive)
       {
          wxMilliSleep(50);
-         pd.Pulse();
+         pd->Pulse();
```

Comment:
Maybe i'm getting ahead of myself, but isn't such interface mean that we do may have UI thread blocking behaviour?

Reply:
That's a very good point.  This function in DBConnection should be called only in the main thread.  I think you can prove it is, from examining the call graph.  (But be cautions about DBConnection -- there are some things in there that happen in other threads).

And it should also be stated that BasicUI::GenericProgressDialog::Pulse() should only called in the main thread.  Likewise for the Update of ProgressDialog.

If there was a mistake (I don't think there was), it was not introduced by this PR.

Reply:
I don't doubt that Pulse is called from the main thread. I'm saying that GenericProgressDialog::Pulse() along with wxMilliSleep blocks execution